error: expected identifier, found keyword `self`
  --> $DIR/ref-self.rs:10:26
   |
10 |     fn take_ref_self(ref self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
   |                          ^^^^ expected identifier, found keyword

error: expected identifier, found keyword `self`
  --> $DIR/ref-self.rs:11:34
   |
11 |     fn take_ref_mut_self(ref mut self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
   |                                  ^^^^ expected identifier, found keyword
